src/cached_digest.ml

```diff
@@ -85,6 +125,7 @@ let file fn =
       needs_dumping := true;
       let stat = Path.stat fn in
       let dirty = ref false in
+      cache.max_timestamp <- Float.max cache.max_timestamp stat.st_mtime;
```

Comment:
Can we extract this and the identical line above into a function? `val set_max_timestamp : cache -> stat -> unit`.

Reply:
Done.